composeApp/src/commonMain/kotlin/sp/bvantur/tasky/register/ui/RegisterScreen.kt

```diff
@@ -37,12 +46,22 @@ import tasky.composeapp.generated.resources.password
 
 @Composable
 fun RegisterRoute() {
-    RegisterScreen()
+    val viewModel = koinViewModel<RegisterViewModel>()
+    val viewState: RegisterViewState by viewModel.viewStateFlow.collectAsStateWithLifecycle()
+
+    RegisterScreen(
+        viewState = viewState,
+        onUserAction = viewModel::onUserAction
+    )
 }
 
 @OptIn(ExperimentalComposeUiApi::class)
 @Composable
-fun RegisterScreen() {
+fun RegisterScreen(viewState: RegisterViewState, onUserAction: OnRegisterUserAction) {
+    val name = rememberSaveable { mutableStateOf("") }
+    val email = rememberSaveable { mutableStateOf("") }
+    val password = rememberSaveable { mutableStateOf("") }
```

Comment:
With the latest Compose version, the TextFields are save AFAIK, since they're based on the new `BasicTextField2` composable. That gives you `TextFieldState` you could put in your VM state and work with it there.
Alternatively, you could put these states as Compose states in your ViewModel.